unittests/svnn_ibc_tests.cpp

```diff
@@ -18,6 +18,46 @@ using namespace eosio::testing;
 
 using mvo = mutable_variant_object;
 
+std::string binary_to_hex(const std::string& bin) {
+   size_t pad_bin_length = (4 - (bin.length() % 4)) % 4;
+
+    std::string padded_bin = std::string(pad_bin_length, '0') + bin;
+    std::stringstream hex_stream;
+
+    for (size_t i = 0; i < padded_bin.length(); i += 4) {
+        std::string segment = padded_bin.substr(i, 4);
+        int value = std::stoi(segment, nullptr, 2);
+        hex_stream << std::hex << value;
+    }
+
+    size_t pad_out_length = (2 - (hex_stream.str().length() % 2)) % 2;
+
+    std::string padded_hex_stream = std::string(pad_out_length, '0') + hex_stream.str();
+    std::string reversed_hex = "";
+
+    for (size_t i = 0; i < padded_hex_stream.length(); i += 2) {
+      std::string byte = padded_hex_stream.substr(i, 2);
+      reversed_hex = byte + reversed_hex;
+    }
+
+    return reversed_hex;
+
+}
+
+std::string bitset_to_binary(const boost::dynamic_bitset<unsigned char>& bitset) {
+   std::string result;
+   result.reserve(bitset.size());
+   for (std::size_t i = bitset.size(); i > 0; --i) {
+       result += bitset[i - 1] ? '1' : '0';
+   }
+   return result;
+}
+
+std::string bitset_to_input_string(const boost::dynamic_bitset<unsigned char>& bitset) {
+   std::string result = bitset_to_binary(bitset);
+   return binary_to_hex(result);
+}

```

Comment:
Probably it doesn't matter, but that's a pretty inefficient way to implement `bitset_to_input_string`. For fun, I implemented a version which converts directly from the bitset to an hex string (and I checked it provides exactly the same result as the existing version): 

```
std::string bitset_to_input_string(const boost::dynamic_bitset<unsigned char>& bitset) {
   static const char* hexchar = "0123456789abcdef";

   boost::dynamic_bitset<unsigned char> bs(bitset);
   bs.resize((bs.size() + 7) & ~0x7);
   assert(bs.size() % 8 == 0);

   std::string result;
   result.resize(bs.size() / 4);
   for (size_t i = 0; i < bs.size(); i += 4) {
      size_t x = 0;
      for (size_t j = 0; j < 4; ++j)
         x += bs[i+j] << j;
      auto slot = i / 4;
      result[slot % 2 ? slot - 1 : slot + 1] = hexchar[x]; // flip the two hex digits for each byte
   }
   return result;
}
```